### DISCUSSION TITLE
Allow VLAN and Prefixes to be attached to site-groups

### DISCUSSION THREAD

**Discussion by empusas** [netbox-community/netbox#8619] [Ideas]
We have a lot of environments that stretch between multiple sites. There are plenty of technologies out there today that support the extension of a layer 2 domain across multiple sites. EVPN with VXLAN, Cisco ACI multi pod, Ethernet over MPLS or pseudo wires.
Whatever the technology is and what pro and cons it may have. The reality is that VLAN and the prefixes attached to them are stretched across multiple sites today.

One way to get around it, is to just create one site in Netbox, but that has a lot of disadvantages as you loose a lot of functionality. The other way is to create multiple sites and then pick one a primary to attach all VLAN and prefixes to.

Both are really just hacks and have a lot of disadvantages. 
Why not make it right and have the reality reflected in Netbox?

**Comment by candlerb** [comment]
> One way to get around it, is to just create one site in Netbox, but that has a lot of disadvantages

When configuring a VLAN, you can decide not to allocate it to a site - then it becomes global, and visible in all sites.

More recently, Netbox also added VLAN Groups and Site Groups. I've not used them myself and I can't comment on whether they'll meet your requirement, but you should check them out.

**Reply by empusas** [reply]
I think using no sites is even worse than having just one site with all VLANs. Then you have no information what VLAN/Prefix is available in which sites.
The sites and site-group should support va similar hierarchy as the regions do.

**Comment by IdRatherStand** [comment]
We also have this issue. As you have mentioned there are several different ways that a VLAN (+ associated prefix(es)) can be present across multiple sites. Would it make more sense to allow a one-to-many relationship within the DB?

By using Site Groups + Vlan Groups we have _sort of_ worked around the problem, however again we become hamstrung in that if we extend a VLAN belonging to a site in a sitegroup to a non-sitegroup site we need to go and create further sitegroups.

**Reply by empusas** [reply]
Sorry for bringing this up again, but to me this is still a big issue.

For VLAN you can work with VLAN groups and assign the scope to a site-group. It is a work around, but we can live with it.

But for prefixes this is really annoying. And it gets even worse now with more and more stuff moving to cloud and IP ranges distributed over eben more data centers.

And no, it is not a solution to just create a site that represents all data centers that share prefixes as you need to keep the site specific information where a physical device is located to ship stuff or send out technicians.
